aws/resource_aws_cognito_user_pool.go

```diff
@@ -536,6 +535,37 @@ func resourceAwsCognitoUserPool() *schema.Resource {
 					},
 				},
 			},
+			"account_recovery_setting": {
+				Type:     schema.TypeList,
+				Optional: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"recovery_mechanisms": {
```

Comment:
As this is a set of structures rather than of simple strings, rename to `recovery_mechanism`.